omnigan/trainer.py

```diff
@@ -1475,11 +986,19 @@ def update_c(self, multi_domain_batch):
 
         """
         zero_grad(self.C)
-        c_loss = self.get_classifier_loss(multi_domain_batch)
-        # ? Log policy
+        if self.opts.train.amp:
+            with autocast():
+                c_loss = self.get_classifier_loss(multi_domain_batch)
+            self.grad_scaler_c.scale(c_loss).backward()
+            self.grad_scaler_c.step(self.c_opt)
+            self.grad_scaler_c.update()
+        else:
+            c_loss = self.get_classifier_loss(multi_domain_batch)
+            self.logger.losses.classifier = c_loss.item()
+            c_loss.backward()
+            self.c_opt_step()
+
         self.logger.losses.classifier = c_loss.item()
-        c_loss.backward()
-        self.c_opt_step()
 
     def get_classifier_loss(self, multi_domain_batch):
```

Comment:
For consistency, we could call this get_C_loss? Even though we're not using it. Also, maybe just before merging (so it does not confuse anyone during PR review), we could change the order of the functions in the trainer? get_G_loss is defined before update_G, while update_D is defined before get_D_loss. We could class functions from the more general to the least general (e.g. train before run_epoch, update_* before get_*_loss), for clarity. Just a sugggestion, don't know if you agree or find this important. :)

Reply:
Or I guess we could do this in another PR

Reply:
You're right, I'll do that